### PR TITLE
pep8: do not exclude directories when they have a parent

### PR DESCRIPTION
**Pull request by stdweird**
Fixes recursive pep8 testing

### CODE REVIEW

**Comment by carlio** [comment]
Anyone know how much the code coverage increased by? We should ask coveralls ;-)

File: prospector/tools/pep8/__init__.py [hunk 1]
```diff
@@ -82,10 +82,11 @@ def excluded(self, filename, parent=None):
 
         # If the file survived pep8's exclusion rules, check it against
         # prospector's patterns.
-        if os.path.isdir(os.path.join(self._files.rootpath, filename)):
+        fullpath = os.path.join(*filter(None, [self._files.rootpath, parent, filename]))
```
Comment (carlio) [inline prospector/tools/pep8/__init__.py]: Could you change this to be a bit more explicit with a list comprehension? `filter(None....` is a bit of an obscurity so I'd prefer the code to be a bit more clear.

Comment (stdweird) [reply prospector/tools/pep8/__init__.py]: @carlio is it better or worse now?

Comment (carlio) [reply prospector/tools/pep8/__init__.py]: @stdweird  I definitely prefer that way, just a personal preference really!

Comment (stdweird) [reply prospector/tools/pep8/__init__.py]: i meant i changed the code, can you check again or do you still want list comprehension?

Comment (carlio) [reply prospector/tools/pep8/__init__.py]: @stdweird yes I saw the change [here for reference](https://github.com/PyCQA/prospector/pull/323/commits/542bd69d8fe6d377f3e98e39e0e41d8306578b8b#diff-eb4112788a404dd185540d70254a98b1R85) it just GitHub hasn't updated this review. I think it's fine now, that's what I meant!

**Comment by stdweird** [comment]
@carlio thanks for merging this. what is the release policy for prospector (and/or an ETA for the next release)?

**Comment by carlio** [comment]
@stdweird I need to find some time myself, there are a few outstanding issues, @chocoelho do you have any time in the near future? 

Basically, ETA is 'unknown' at this point.

**Pull request merged**